src/ProxyClient/Fastly.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\Exception\InvalidArgumentException;
+use FOS\HttpCache\ProxyClient\Invalidation\BanCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\RefreshCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Fastly HTTP cache invalidator.
+ *
+ * @author Simone Fumagalli <simone.fumagalli@musement.com>
+ */
+class Fastly extends HttpProxyClient implements TagCapable
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function invalidateTags(array $tags)
+    {
+
+        $headers = [
+            'Fastly-Key' => $this->options['authentication_token'],
+            'Accept' => 'application/json'
+        ];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        foreach (\explode(",", $this->options['service_identifier']) as $fastlyServiceId) {
+            foreach ($tags as $tag) {
```

Comment:
In the case of hard purge, it's possible to invalidate 256 tags at a time now:
https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583

Reply:
Seems they are also able to do this on Soft purge, via support there they seem to plan to update doc to make it more clear.

So given Faslty has API rate limits, we should change to use the API linked to above regardless of soft/purge so we can do up to 256 at a time.